### ISSUE TITLE
Self-sizing cell example

### ISSUE THREAD

**Issue by rnystrom** [Instagram/IGListKit#149]
Both #80 and #143 show that self-sizing cells with `IGListKit` is a little hard to follow. I think we need to get a working example. Some things that would be great to see:

- Setting up self-sizing cells **in code**
- Using storyboards

**Comment by rotemdoron** [comment]
hi guys we are working on a new project that really needs self sizing cells, any working example somwhere yet?

**Comment by akaralar** [comment]
I have the same problem, on first launch the cells are animating from zero size to current size. I was using `layout.estimatedItemSize = UICollectionViewFlowLayoutAutomaticSize` and laying out cells with auto layout. Later I created dummy cells to get the height of the cells when they are laid out and return that in `sizeForItem(at index:)` but still initial animation persists.

**Comment by rnystrom** [comment]
@rotemdoron @akaralar check out #176, got it working with both nibs and just code. Hopefully this helps!

**Comment by rotemdoron** [comment]
hi rnystrom thank you! the problam is we are waiting for data from a web service as you probably do. so we do the performUpdates which with self sizing cells make the collectionview very "jumpy" animating all cells at once and animating from 0 to the size in first load. any idea's how to solve this? (happens also without performUpdates with animate, still jump just without the animation)

**Comment by rnystrom** [comment]
@rotemdoron Hmm, that's an interesting example. First simple idea is to do `adapter.performUpdates(animated: false)` if its the first load?

**Comment by rotemdoron** [comment]
@rnystrom still happens just without the animation, it jumps to the size instead of animating to it also looks bad. performUpdates updated the entire list though and not only the "diffable" changed object, when using estimatedSize i mean.

**Comment by rnystrom** [comment]
@rotemdoron that sort of sounds expected though since you start w/out any data and then go to having data. Is there a chance you could provide an example setup that has the behavior you're seeing?

**Comment by rotemdoron** [comment]
basicly the data array is empty untill data is fetched and when we performUpdates (with self sizing) the entire list jumps in to the size of the cell, it is expected from a collectionview but usually the "jump" is not so ruff, becouse you can use reloadData and not batch update which animate everything.
 
 func objects(for listAdapter: IGListAdapter) -> [IGListDiffable] {
        return dataArray
    }

**Comment by rnystrom** [comment]
@rotemdoron but if you use `adapter.performUpdates(animated: false)` (or just `adapter.reloadData()` the update is the same as vanilla collection view, right?

**Comment by rotemdoron** [comment]
@rnystrom  adapter.reloadData() makes the layout messy, i dont know why, basically only when im setting sizeForItem and calculating the size my self, its works good, self sizing cells not working so good after performUpdate() (animated or not) and reloadData()

**Comment by BestKai** [comment]
I have the same problem,but i dont known how to solve it
